java/ql/src/experimental/Security/CWE/CWE-665/InsecureRmiJmxEnvironmentConfiguration.ql

```diff
@@ -0,0 +1,146 @@
+/**
+ * @name InsecureRmiJmxAuthenticationEnvironment
+ * @description This query detects if a JMX/RMI server is created with a potentially dangerous environment, which could lead to code execution through insecure deserialization.
+ * @kind path-problem
+ * @problem.severity error
+ * @tags security
+ *       external/cwe/cwe-665
+ * @precision high
+ * @id java/insecure-rmi-jmx-server-initalisation
+ */
+
+import java
+import semmle.code.java.dataflow.DataFlow
+import semmle.code.java.dataflow.DataFlow2
+import semmle.code.java.Maps
+import DataFlow::PathGraph
+import semmle.code.java.dataflow.NullGuards
+import semmle.code.java.dataflow.Nullness
+
+/** predicate which detects vulnerable Constructors */
+predicate isRmiOrJmxServerCreateConstructor(Constructor constructor) {
+  constructor.getName() = "RMIConnectorServer" and
+  constructor
+      .getDeclaringType()
+      .hasQualifiedName("javax.management.remote.rmi", "RMIConnectorServer")
+}
+
+/** Predicate which detects vulnerable server creations via methods */
+predicate isRmiOrJmxServerCreateMethod(Method method) {
+  method.getName() = "newJMXConnectorServer" and
+  method.getDeclaringType().hasQualifiedName("javax.management.remote", "JMXConnectorServerFactory")
+}
+
+/**
+ * Models flow from `new HashMap<>()` to a
+ * `map.put("jmx.remote.rmi.server.credential.types", value)` call.
+ */
+class MapToPutCredentialstypeConfiguration extends DataFlow2::Configuration {
+  MapToPutCredentialstypeConfiguration() { this = "MapToPutCredentialstypeConfiguration" }
+
+  override predicate isSource(DataFlow::Node source) {
```

Comment:
If I'm completely honest I couldn't tell if this makes a difference (function wise) (I'm happy to learn though...).
But if it makes no difference, then yes it would be better to change it to DataFlow2